x/mongo/driver/connstring/connstring.go

```diff
@@ -340,6 +342,19 @@ func (p *parser) validate() error {
 		}
 	}
 
+	// Validation for load-balanced mode.
+	if p.LoadBalancedSet && p.LoadBalanced {
+		if len(p.Hosts) > 1 {
+			return internal.ErrLoadBalancedWithMultipleHosts
+		}
+		if p.ReplicaSet != "" {
+			return internal.ErrLoadBalancedWithReplicaSet
+		}
+		if p.ConnectSet || p.DirectConnectionSet {
```

Comment:
What's the difference between `Connect` and `DirectConnection` on a Connstring?

Reply:
They serve the same purpose. `Connect` is a legacy Go-only URI option with values `automatic` (default) and `direct`, which allows you to create a direct connection to a node. In the 4.4 server release cycle, we standardized this across drivers by adding a new `directConnection` option with values `true` and `false` (default). In `clientoptions.go` we document that the `connect` option is deprecated in favor of `directConnection`, but until we do a major version bump, we need to account for both.

Reply:
Ah ok, thanks for the context!